src/Common/src/System/Resources/ResourceWriter.cs

```diff
@@ -236,8 +224,8 @@ private void Dispose(bool disposing)
                 }
             }
 
-            _output = null;
-            _caseInsensitiveDups = null;
+            _output = null!;
```

Comment:
In the common case like here where the pattern is null iff disposed, it makes sense to me to bang the assignments in dispose. That way all the rest of the code can get the benefits of the null checking, and no code should run after dispose -- indeed checks are made against null to enforce that.